### ISSUE TITLE
The 3.4.1 version is slowing down my full machine, looking at antora.xml with ripgrep in a large folder

### ISSUE THREAD

**Issue by loic-joly-sonarsource** [asciidoctor/asciidoctor-vscode#921]
My working folder contains thousands of subfolders. In one of the subfolders, I edit an asciidoc file. Each time I perform any edit, the 32 cores of my computer (on Windows) jump to 100% usage for several seconds. 

![image](https://github.com/user-attachments/assets/78292a1d-002f-4bcf-80c7-99cda15366b1)

When looking at the greedy process, I notice rg.exe, which is launched with the following command line:
````
"c:\Users\****\AppData\Local\Programs\Microsoft VS Code\resources\app\node_modules\@vscode\ripgrep\bin\rg.exe" --hidden --follow --files -g **/antora.yml
````

Going back to version 3.1.1 solves the issue.

**Comment by ggrossetie** [comment]
In theory, `ripgrep` should be faster (and consume less resources than `vscode.workspace.findFiles`. Could you please share some stats about your large folder? Do you have symlinks? How many files do you have? How many hidden files? How many `antora.yml`? How many directories?

If you manually run `"c:\Users\****\AppData\Local\Programs\Microsoft VS Code\resources\app\node_modules\@vscode\ripgrep\bin\rg.exe" --hidden --follow --files -g **/antora.yml` on your workspace folder, how long does it take?

**Comment by ggrossetie** [comment]
Fixed in c854814. I'm keeping this issue open because at some point we might re-enable ripgrep and I want to make sure that it will not impact performance on some cases.

**Comment by loic-joly-sonarsource** [comment]
This folder contains 21700 files, evenly spread out across 10700 directories. It does not contain any file named `antora.yml`, only one hidden folder (`.git`), and no symlink. 
When I run the command manually, it takes ~2s. When I had the issue, I noticed there were several instances of that command running in parallel.

Is there any other info that might help you?

**Comment by loic-joly-sonarsource** [comment]
Switching to 3.4.2 mostly solves the issue. I still have CPU spikes, but they are much smaller.

Going back to 3.4.1:
* I looked at several rg runs, they were all about `antora.yml`.
* It's hard to count, but about 7 `rg.exe` processes were started with each keystroke.
* Disabling Antora support works great, no spike at all.

I now went back to 3.4.2, with no support for Antora.

**Comment by aallrd** [comment]
I have observed this issue as well in a large monorepo: hundreds of ripgrep processes spawning that make my machine unusable when Antora support is enabled in the extension (3.4.4):
<img width="793" height="1336" alt="Image" src="https://github.com/user-attachments/assets/a4d0d07e-3131-4fe8-b6de-79f3b50d46bc" />
<img width="748" height="455" alt="Image" src="https://github.com/user-attachments/assets/9cefe212-1479-4b15-8ac8-29da2f0fedb0" />

All the `rg` processes are the same:
```
"C:\Users\aallard\AppData\Local\Programs\Microsoft VS Code\resources\app\node_modules\@vscode\ripgrep\bin\rg.exe" --files --hidden --case-sensitive --no-require-git -g **/antora.yml -g !**/.git -g !**/.svn -g !**/.hg -g !**/.DS_Store -g !**/Thumbs.db --no-ignore --follow --no-config --no-ignore-global
```

It's not clear to me why so many processes are spawned.

All my Antora documentation is under a _docs/_ folder, I don't need the whole workspace to be scanned (that's the root directory for the `rg` processes).
If I open my _docs/_ folder in VSCode instead of the root of my repo, then everything works fine.

Is there a way to limit this Antora support/scan to a specific directory ? Maybe automatically set it to the directory where the _antora.yml_ file is found ?

**Comment by ggrossetie** [comment]
Performance improvements are coming in the next major version 4.0.
@aallrd @loic-joly-sonarsource Please open a new issue if you still have performance issues with the upcoming version.
